### ISSUE TITLE
Does it support postgresql database backup?

### ISSUE THREAD

**Issue by Seele0oO** [ttionya/vaultwarden-backup#88]
Use this parameter.
```yaml
DATABASE_URL: "postgresql://exampleuser:examplepasswd@db/postgres"
```
The vault can be stored to it. Using this approach you can create distributed vaultwarden.

So does this backup function support postgresql backup function?

**Comment by ttionya** [comment]
@Seele0oO ,

I have researched and only need to modify a small amount of code to support PostgreSQL and MariaDB/MySQL .

But I'm not good with databases and **don't know how to backup and restore PostgreSQL and MariaDB/MySQL data properly**.

Another problem occurs in the restore phase, sqlite can be restored by just copying the db file to the folder, but other databases need to execute sql, this is also the aspect that bothers me.

If you can provide the shell command for PostgreSQL database backup and restore, I will finish the support for PostgreSQL as soon as possible. If you can also provide commands for MariaDB/MySQL, that would be great.

**Comment by Seele0oO** [comment]
Thank you for your attention,I have done some shell scripts, but I am not very skilled, so I wrote some example, hope it is useful for you.

https://github.com/Seele0oO/vaultwarden-backup/commit/b9e1c2496522fb264497b9822d810b1649a8a2c9

**Comment by Seele0oO** [comment]
by the way,It's not a good idea to use mysql, I think it's too heavy.
There is some example.
https://github.com/Seele0oO/vaultwarden-backup/commit/42272732db698716075e841ee96825d7b8f1f728

In my view,the MariaDB/MySQL is the same in commands.
So it's probably universal.

**Comment by ttionya** [comment]
@Seele0oO ,

The PostgreSQL backend is now supported, try it on `1.16.0-beta.0`.

Please let me know if you have any questions.

I will continue to add MySQL/MariaDB support.

**Comment by Seele0oO** [comment]
thanks for your work.I will test it in few days later.Take care yourself and be well. It is the time to sleeping.Sent from my iPhoneOn 11 Dec 2022, at 23:34, ttionya ***@***.***> wrote:﻿
@Seele0oO ,
The PostgreSQL backend is now supported, try it on 1.16.0-beta.0.
Please let me know if you have any questions.
I will continue to add MySQL/MariaDB support.

**Comment by ttionya** [comment]
`1.16.0-beta.1` is support Mysql(MariaDB) and PostgreSQL bankend.

**Comment by ttionya** [comment]
`1.16.0-beta.2` is support PostgreSQL 15. And earlier beta versions are no longer available.

**Comment by Seele0oO** [comment]
I'm wondering if I'm missing some environment variables that specify the backup method? In my tests, the data in the datagrip-connected postgresql database changed as I made changes, but the backup program still just backed up those files.
![image](https://user-images.githubusercontent.com/48525679/209542315-b2e600e5-ecb7-4150-b5c0-0c035b202102.png)
After I added some attachments to my test account, these two folders remained empty.
![image](https://user-images.githubusercontent.com/48525679/209542545-1f08d118-36d5-4db8-8429-abbe32da8c89.png)

**Comment by ttionya** [comment]
Please provide environment variables.

**Comment by Seele0oO** [comment]
```yml
vaultwarden:
  environment:
     SIGNUPS_ALLOWED: 'true'
     DATABASE_URL: 'postgresql://postgres:example@db/postgres'
```

```yml
backup:
    environment:
      CRON: '*/5 * * * *'
```

```yml
  db:
    image: postgres
    restart: always
    environment:
      POSTGRES_PASSWORD: example
```

**Comment by ttionya** [comment]
Sorry, please check the [document](https://github.com/ttionya/vaultwarden-backup/pull/89/files#diff-f1e2e89eae2c892d04bc0031919e83f293cd23984e2b0a21315642d4da348950).

**Comment by ttionya** [comment]
@Seele0oO ,

Hello, is the postgresql database backup working?

**Comment by Seele0oO** [comment]
It works fine on Debain 11.
I can't test it with mysql because of the preformance of my password-server.

**Comment by ttionya** [comment]
Released stable version v1.16.0.
